### ISSUE TITLE
open with :Via(module) always returns 1

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#5681]
Migrated from [rt.perl.org#9992](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=9992) (status was 'resolved')

Searchable as RT9992$

**Comment by p5pRT** [comment]
### From @lizmat

#### Created by @lizmat



== ioopenpb ==========================================================
use MIME&#8203;::QuotedPrint;
my $return = open\( my $in\,'\<&#8203;:Via\(MIME&#8203;::QuotedPrint\)'\, 'doesnotexist' \);
warn "return = $return\, in = $in";
======================================================================

produces the following output&#8203;:

$ perl ioopenpb
return = 1\, in = GLOB\(0x8125b94\) at ioopenpb line 3\.
Can't use an undefined value as a symbol reference at 
/usr/local/lib/perl5/5\.8\.0
/i686\-linux\-thread\-multi/MIME/QuotedPrint\.pm line 184\.
Can't use an undefined value as a symbol reference at 
/usr/local/lib/perl5/5\.8\.0
/i686\-linux\-thread\-multi/MIME/QuotedPrint\.pm line 184\.

Seems like PerlIO&#8203;::Via is not properly checking whether the file could be
opened or not\.  This problem only seems to occur with PerlIO&#8203;::Via\.

<details><summary>Perl Info</summary>

```

Flags:
     category=core
     severity=high

Site configuration information for perl v5.8.0:

Configured by liz at Fri Jun 21 19:59:52 CEST 2002.

Summary of my perl5 (revision 5.0 version 8 subversion 0 patch 17339) 
configuration:
   Platform:
     osname=linux, osvers=2.4.18, archname=i686-linux-thread-multi
     uname='linux echt.dijkmat.nl 2.4.18 #8 smp mon mar 25 22:28:36 cet 
2002 i686 unknown '
     config_args='-de -Dusethreads'
     hint=recommended, useposix=true, d_sigaction=define
     usethreads=define use5005threads=undef useithreads=define 
usemultiplicity=define
     useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
     use64bitint=undef use64bitall=undef uselongdouble=undef
     usemymalloc=n, bincompat5005=undef
   Compiler:
     cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing 
-I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 
-I/usr/include/gdbm',
     optimize='-O2',
     cppflags='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing 
-I/usr/local/include -I/usr/include/gdbm'
     ccversion='', gccversion='2.96 20000731 (Red Hat Linux 7.1 2.96-98)', 
gccosandvers=''
     intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
     d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
     ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', 
lseeksize=8
     alignbytes=4, prototype=define
   Linker and Libraries:
     ld='cc', ldflags =' -L/usr/local/lib'
     libpth=/usr/local/lib /lib /usr/lib
     libs=-lnsl -lgdbm -ldb -ldl -lm -lpthread -lc -lcrypt -lutil
     perllibs=-lnsl -ldl -lm -lpthread -lc -lcrypt -lutil
     libc=/lib/libc-2.2.4.so, so=so, useshrplib=false, libperl=libperl.a
     gnulibc_version='2.2.4'
   Dynamic Linking:
     dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
     cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
     DEVEL17339


@INC for perl v5.8.0:
     /usr/local/lib/perl5/5.8.0/i686-linux-thread-multi
     /usr/local/lib/perl5/5.8.0
     /usr/local/lib/perl5/site_perl/5.8.0/i686-linux-thread-multi
     /usr/local/lib/perl5/site_perl/5.8.0
     /usr/local/lib/perl5/site_perl/5.7.3
     /usr/local/lib/perl5/site_perl
     .


Environment for perl v5.8.0:
     HOME=/home/liz
     LANG=en_US
     LANGUAGE (unset)
     LD_LIBRARY_PATH (unset)
     LOGDIR (unset)
     PATH=/usr/local/bin:/bin:/usr/bin:/usr/X11R6/bin:/home/liz/bin
     PERL_BADLANG (unset)
     SHELL=/bin/bash




```
</details>

**Comment by p5pRT** [comment]
### From @jhi

This problem was resolved by perlio change \#17795\, which got into bleadperl as change \#18086\, and maintperl as change \#18095\.  I'm marking the problem ticket as resolved\.

**Comment by p5pRT** [comment]
@jhi - Status changed from 'open' to 'resolved'
